Add Azure gems to README

ffmike opened the pull request.

Commit:
Add Azure gems to README

georgeclaghorn commented on activestorage/README.md:
```diff
@@ -88,7 +88,8 @@ Variation of image attachment:
 2. Optional: Add `gem "aws-sdk", "~> 2"` to your Gemfile if you want to use AWS S3.
 3. Optional: Add `gem "google-cloud-storage", "~> 1.3"` to your Gemfile if you want to use Google Cloud Storage.
-4. Optional: Add `gem "mini_magick"` to your Gemfile if you want to use variants.
+4. Optional: Add `gem 'azure-core', git: "https://github.com/dixpac/azure-ruby-asm-core.git"` and `gem 'azure-storage', require: false` to your Gemfile if you want to use Microsoft Azure.
```
Minor nit: use double quotes to match the other optional steps.

ffmike replied:
Yeah, good point.

Also at some point presumably we can switch to the release version of the azure-core gem. I'm torn between wanting to note that or not, it's one more thing to keep track of for the future.

georgeclaghorn replied:
Looks like @dixpac’s PR (Azure/azure-ruby-asm-core#39) was merged. @dixpac, can we switch back to the released gem now?

Commit:
Update README.md

schneems commented on activestorage/README.md:
```diff
@@ -88,7 +88,8 @@ Variation of image attachment:
 2. Optional: Add `gem "aws-sdk", "~> 2"` to your Gemfile if you want to use AWS S3.
 3. Optional: Add `gem "google-cloud-storage", "~> 1.3"` to your Gemfile if you want to use Google Cloud Storage.
-4. Optional: Add `gem "mini_magick"` to your Gemfile if you want to use variants.
+4. Optional: Add `gem "azure-core", git: "https://github.com/dixpac/azure-ruby-asm-core.git"` and `gem "azure-storage", require: false` to your Gemfile if you want to use Microsoft Azure.
```
Why doesn't the released version work here? Any idea when master will be released in a new gem?

georgeclaghorn replied:
The released gem didn't support StringIO uploads. It does as of yesterday.

ffmike replied:
Just confirmed in my sample project that the release gem is working fine. Updated the PR accordingly.

Commit:
Use release azure-core gem

kamipo commented on activestorage/README.md:
```diff
@@ -88,7 +88,8 @@ Variation of image attachment:
 2. Optional: Add `gem "aws-sdk", "~> 2"` to your Gemfile if you want to use AWS S3.
 3. Optional: Add `gem "google-cloud-storage", "~> 1.3"` to your Gemfile if you want to use Google Cloud Storage.
-4. Optional: Add `gem "mini_magick"` to your Gemfile if you want to use variants.
+4. Optional: Add `gem "azure-core"` and `gem "azure-storage", require: false` to your Gemfile if you want to use Microsoft Azure.
```
We don't need to add `gem "azure-core"` once #30041 has merged.

georgeclaghorn commented on activestorage/README.md:
```diff
@@ -88,7 +88,7 @@ Variation of image attachment:
 2. Optional: Add `gem "aws-sdk", "~> 2"` to your Gemfile if you want to use AWS S3.
 3. Optional: Add `gem "google-cloud-storage", "~> 1.3"` to your Gemfile if you want to use Google Cloud Storage.
-4. Optional: Add `gem "azure-core", git: "https://github.com/dixpac/azure-ruby-asm-core.git"` and `gem "azure-storage", require: false` to your Gemfile if you want to use Microsoft Azure.
+4. Optional: Add `gem "azure-core"` and `gem "azure-storage", require: false` to your Gemfile if you want to use Microsoft Azure.
```
The  `azure-storage` gem depends on `azure-core`, so we need not manually bundle the latter.

georgeclaghorn replied:
Let's omit `require: false` to match the other optional steps. It's not strictly necessary.

Commit:
Tweak requires

Omit azure-core (it's required by azure-storage), remove require:false to match other points.

dixpac commented:
We can switch to the official release as of yesterday. I've submitted the PR yesterday #30041

schneems commented:
Great, thanks!
